clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3560,18 +3560,10 @@ static void handleSYCLIntelNumSimdWorkItemsAttr(Sema &S, Decl *D,
 }
 
 // Handles use_stall_enable_clusters
-static void handleUseStallEnableClustersAttr(Sema &S, Decl *D,
-                                             const ParsedAttr &Attr) {
-  if (D->isInvalidDecl())
-    return;
-
-  unsigned NumArgs = Attr.getNumArgs();
-  if (NumArgs > 0) {
-    S.Diag(Attr.getLoc(), diag::warn_attribute_too_many_arguments) << Attr << 0;
-    return;
-  }
-
-  handleSimpleAttribute<SYCLIntelUseStallEnableClustersAttr>(S, D, Attr);
+static void handleSYCLIntelUseStallEnableClustersAttr(Sema &S, Decl *D,
+                                                      const ParsedAttr &A) {
+  D->addAttr(::new (S.Context)
```

Comment:
Are the deleted checks done elsewhere already?

Reply:
Not quite i understand your question. Did you mean this checl?
```
unsigned NumArgs = Attr.getNumArgs();
  if (NumArgs > 0) {
    S.Diag(Attr.getLoc(), diag::warn_attribute_too_many_arguments) << Attr << 0;
    return;
  }``

Existing test (SemaSYCL/stall_enable_device.cpp) shows error for wrong attribute argument. So this check is useless.
          the behavior of automatic diagnostic checking when passing
          the wrong number of arguments

          Example:
          [[intel::use_stall_enable_clusters(1)]] void test1() {} 
          // expected-error{{'use_stall_enable_clusters' attribute takes no arguments}}

Reply:
> 

Yes, they're handled in `ProcessDeclAttribute()` when calling `S.checkCommonAttributeFeatures(D, AL))`.

Reply:
Ok. Thanks for clarifying!